### PR TITLE
Mate HW 8 Soution added

### PR DESCRIPTION
**Pull request by AlexanderSashchenko**

### CODE REVIEW

File: src/main/java/core/basesyntax/MyLinkedList.java [hunk 1]
```diff
@@ -1,50 +1,151 @@
...
-
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("Wrong index " + index);
+        }
+        if (index == size) {
+            add(value);
+            return;
+        }
+        Node<T> previous = findNode(index).prev;
+        Node<T> next = findNode(index);
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/MyLinkedList.java]: try using method findNode() only one time

File: src/main/java/core/basesyntax/MyLinkedList.java [hunk 2]
```diff
@@ -1,50 +1,151 @@
...
+        } else {
+            next.prev = current;
+        }
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (T t : list) {
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/MyLinkedList.java]: `t` isn't informative name of the variable

File: src/main/java/core/basesyntax/MyLinkedList.java [hunk 3]
```diff
@@ -1,50 +1,151 @@
...
+        } else {
+            previous.next = next;
+        }
+        if (next == null) {
+            tail = previous;
+        } else {
+            next.prev = previous;
+        }
+        size--;
+        return toRemove.value;
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/MyLinkedList.java]: how about removing the link on previous and next node?

File: src/main/java/core/basesyntax/MyLinkedList.java [hunk 4]
```diff
@@ -1,50 +1,151 @@
...
+            throw new IndexOutOfBoundsException("Wrong index " + index);
+        }
+        Node<T> toFind;
+        if (index < size / 2) {
+            toFind = head;
+            for (int i = 0; i < index; i++) {
+                toFind = toFind.next;
+            }
+        }
+        toFind = tail;
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/MyLinkedList.java]: add else block

File: src/main/java/core/basesyntax/MyLinkedList.java [hunk 5]
```diff
@@ -1,50 +1,151 @@
...
+        }
+        return toFind;
+    }
+
+    private static class Node<T> {
+        private T value;
+        private Node<T> next;
+        private Node<T> prev;
+
+        Node(Node<T> prev, T value, Node<T> next) {
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/MyLinkedList.java]: add access modifier

File: src/main/java/core/basesyntax/MyLinkedList.java (hunk 2 above)
Comment (AlexanderSashchenko) [reply src/main/java/core/basesyntax/MyLinkedList.java]: Fixed.

File: src/main/java/core/basesyntax/MyLinkedList.java (hunk 4 above)
Comment (AlexanderSashchenko) [reply src/main/java/core/basesyntax/MyLinkedList.java]: Fixed.

File: src/main/java/core/basesyntax/MyLinkedList.java (hunk 5 above)
Comment (AlexanderSashchenko) [reply src/main/java/core/basesyntax/MyLinkedList.java]: Fixed.

File: src/main/java/core/basesyntax/MyLinkedList.java (hunk 3 above)
Comment (AlexanderSashchenko) [reply src/main/java/core/basesyntax/MyLinkedList.java]: We discussed it with Bogdan. As soon as nothing links to removed node the Garbage collector removes the node. Propose to let it as it is.

File: src/main/java/core/basesyntax/MyLinkedList.java (hunk 1 above)
Comment (AlexanderSashchenko) [reply src/main/java/core/basesyntax/MyLinkedList.java]: Fixed.

File: src/main/java/core/basesyntax/MyLinkedList.java (hunk 3 above)
Comment (Sofasmile) [reply src/main/java/core/basesyntax/MyLinkedList.java]: ок

**Pull request closed without merging**